crates/rpc/rpc/src/eth/api/pending_block.rs

```diff
@@ -130,11 +131,13 @@ impl PendingBlockEnv {
             }
 
             // Configure the environment for the block.
-            let env =
-                Env { cfg: cfg.clone(), block: block_env.clone(), tx: tx_env_with_recovered(&tx) };
+            let env = Box::new(Env {
+                cfg: cfg.cfg_env.clone(),
+                block: block_env.clone(),
+                tx: tx_env_with_recovered(&tx),
+            });
 
-            let mut evm = revm::EVM::with_env(env);
-            evm.database(&mut db);
+            let mut evm = revm::Evm::builder().modify_env(|e| *e = env).with_db(&mut db).build();
```

Comment:
there should be a function that accepts the env: `fn with_env`

Reply:
added `with_env`, `with_block`, `with_tx`